lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -255,11 +255,45 @@ class StableHLOToTTIRConstantOpConversionPattern
   matchAndRewrite(mlir::stablehlo::ConstantOp srcOp,
                   mlir::stablehlo::ConstantOp::Adaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
+    LogicalResult legalityResult = checkBasicLegality(srcOp, rewriter);
+    if (!legalityResult.succeeded()) {
+      return legalityResult;
+    }
+
     auto outputType = mlir::cast<RankedTensorType>(
         getTypeConverter()->convertType(srcOp.getResult().getType()));
 
+    mlir::ElementsAttr valueAttr = srcOp.getValue();
+    if (valueAttr.getShapedType().getShape().empty()) {
```

Comment:
Should this be an assertion? What happens if it's not a splat?

Reply:
If it's not splat then we will assert in TTIR->TTNN conversion for constant op (see [here](https://github.com/tenstorrent/tt-mlir/blob/main/lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp#L490)). I didn't want to assert here because it is not limitation of StableHLO->TTIR conversion and will be common for all third party dialects, it is TTNN limitation.